### PR TITLE
[many ports] Fix various problems with mingw

### PR DESCRIPTION
**Pull request by longnguyen2004**
**Describe the pull request**

- What does your PR fix? Fixes #
This PR fixes various problems with ports on mingw
- Which triplets are supported/not supported? Have you updated the CI baseline?
All mingw triplets
- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes

### CODE REVIEW

File: ports/sdl2/export-symbols-only-in-shared-build.patch [hunk 1]
```diff
@@ -7,14 +7,16 @@
...
+
 diff -r 8df7a59b5528 -r 46ec9baae30c include/begin_code.h
 --- a/include/begin_code.h	Fri Sep 22 11:25:52 2017 -0700
 +++ b/include/begin_code.h	Sun Sep 24 14:32:30 2017 +0300
 @@ -58,8 +58,10 @@
  #   else
  #    define DECLSPEC    __declspec(dllimport)
  #   endif
-+#  elif defined(_DLL)
++#  elif defined(SDL2_EXPORTS)
```
Comment (JackBoosY) [inline ports/sdl2/export-symbols-only-in-shared-build.patch:19]: Does this macro used in the `CMakeLists.txt`?

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/sdl2/export-symbols-only-in-shared-build.patch (hunk 1 above)
Comment (longnguyen2004) [reply ports/sdl2/export-symbols-only-in-shared-build.patch:19]: I can't find the macro in the `CMakeList.txt` itself, but when I check the build log, it's defined by...something
Also, the other sdl2 accessories in vcpkg use this macro in the custom `CMakeList.txt` (see [sdl2-net](https://github.com/microsoft/vcpkg/blob/master/ports/sdl2-net/CMakeLists.txt#L13), [sdl2-ttf](https://github.com/microsoft/vcpkg/blob/master/ports/sdl2-ttf/CMakeLists.txt#L10), [sdl2-image](https://github.com/microsoft/vcpkg/blob/master/ports/sdl2-image/CMakeLists.txt#L60), [sdl2-mixer](https://github.com/microsoft/vcpkg/blob/master/ports/sdl2-mixer/CMakeLists.txt#L106))

Comment (JackBoosY) [reply ports/sdl2/export-symbols-only-in-shared-build.patch:19]: @longnguyen2004 So I think it's not correct.

Comment (longnguyen2004) [reply ports/sdl2/export-symbols-only-in-shared-build.patch:19]: I'm still trying to find where it's defined. In the build log you can see `-DSDL2_EXPORTS`, but I don't know where it came from. Can someone help me find it out?

Comment (JackBoosY) [reply ports/sdl2/export-symbols-only-in-shared-build.patch:19]: It should be provided by sdl-image, and I think this macro shouldn't be used in sdl2.

Comment (longnguyen2004) [reply ports/sdl2/export-symbols-only-in-shared-build.patch:19]: Hmm I'll take this commit down for now, while I think of a better solution

File: ports/opus/portfile.cmake [hunk 2]
```diff
@@ -14,9 +14,16 @@ vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
...
+if(VCPKG_TARGET_IS_MINGW)
+  set(STACK_PROTECTOR OFF)
+  string(APPEND VCPKG_C_FLAGS "-D_FORTIFY_SOURCE=0")
+  string(APPEND VCPKG_CXX_FLAGS "-D_FORTIFY_SOURCE=0")
+endif()
+
 vcpkg_configure_cmake(
     SOURCE_PATH ${SOURCE_PATH}
     OPTIONS ${FEATURE_OPTIONS}
+    OPTIONS -DOPUS_STACK_PROTECTOR=${STACK_PROTECTOR}
```
[the remark this answers is not in the document]
Comment (longnguyen2004) [reply ports/opus/portfile.cmake]: Should be good now

**Review by JackBoosY — APPROVED** [review]

**Pull request merged**